src/rqt_plot/plot_widget.py

```diff
@@ -33,92 +33,97 @@
-    topics = node.get_topic_names_and_types()
-    real_topic = None
-    for name, topic_types in topics:
-        if name == topic_name[:len(name)]:
-            real_topic = name
-            topic_type_str = topic_types[0] if topic_types else None
-            break
-    if real_topic is None:
+    topic_type, real_topic, _ = get_topic_type(node, topic_name)
+    if topic_type is None:
         message = "topic %s does not exist" % (topic_name)
         return [], message
-
-    if topic_type_str is None:
-        message = "no topic types found for topic %s " % (topic_name)
-        return [], message
-
-    if len(topic_name) < len(real_topic) + 1:
-        message = 'no field specified in topic name "{}"'.format(topic_name)
-        return [], message
-
     field_name = topic_name[len(real_topic) + 1:]
 
-    message_class = message_helpers.get_message_class(topic_type_str)
-    if message_class is None:
-        message = 'message class "{}" is invalid'.format(topic_type_str)
-        return [], message
-
-    slot_type, is_array, array_size = _parse_type(topic_type_str)
+    slot_type, is_array, array_size = _parse_type(topic_type)
     field_class = message_helpers.get_message_class(slot_type)
+    if field_class is None:
+        message = "type of topic %s is unknown" % (topic_name)
+        return [], message
 
+    # Go through the fields until you reach the last msg field
     fields = [f for f in field_name.split('/') if f]
-
     for field in fields:
         # parse the field name for an array index
-        field, _, field_index = _parse_type(field)
-        if field is None:
+        try:
+            field, _, field_index = _parse_type(field)
+        except roslib.msgs.MsgSpecException:
             message = "invalid field %s in topic %s" % (field, real_topic)
             return [], message
 
-        field_names_and_types = field_class.get_fields_and_field_types()
-        if field not in field_names_and_types:
+        if not hasattr(field_class, "get_fields_and_field_types"):
+            msg = "Invalid field path %s in topic %s" % (field_name, real_topic)
+            return [], msg
+
+        fields_and_field_types = field_class.get_fields_and_field_types()
+        if field not in fields_and_field_types.keys() :
             message = "no field %s in topic %s" % (field_name, real_topic)
             return [], message
-        slot_type = field_names_and_types[field]
+
+        slot_type = fields_and_field_types[field]
         slot_type, slot_is_array, array_size = _parse_type(slot_type)
         is_array = slot_is_array and field_index is None
 
-        if topic_helpers.is_primitive_type(slot_type):
-            field_class = topic_helpers.get_type_class(slot_type)
-        else:
-            field_class = message_helpers.get_message_class(slot_type)
+        field_class = message_field_type_helpers.get_type_class(slot_type)
 
+    # TODO: add bytes to this as you could treat bytes as an array of uint
```

Comment:
Byte array is standard type, so I think this should be included and nix the TODO

Reply:
On further consideration, I don't think we should do this. It would be like plotting the first character of a string. Not that useful imho